vunit/test/unit/test_test_suites.py

```diff
@@ -107,3 +108,38 @@ def _read_test_results(contents, expected_test_cases):
                           test_suite_name=None,
                           test_cases=expected_test_cases)
             return run._read_test_results(file_name=file_name)  # pylint: disable=protected-access
+
+    def test_exit_code(self):

```

Comment:
I am missing some comment explaining what is the purpose.
I am also missing a test that checks that SKIPPED results are not modified.
I am also missing a test with two result where one is PASSED and one is FAILED and the results are unmodified.

Reply:
> I am missing some comment explaining what is the purpose.

I will ad a comment.

> I am also missing a test that checks that SKIPPED results are not modified.

Do you mean that `passed, skipped` + `exit_code != 0` should produce `failed, skipped`?

Also, regarding the SKIPPED status:

> Would it be acceptable to allow PASSED and/or SKIPPED to be passed to call_post_check instead?

At the moment, `call_post_check` is only executed if all the tests PASSED.

> I am also missing a test with two result where one is PASSED and one is FAILED and the results are unmodified.

I will add them too.

Reply:
> Do you mean that `passed, skipped` + `exit_code != 0` should produce `failed, skipped`?

Yes. Or passed, failed, skipped => failed, failed, skipped

> 
> Also, regarding the SKIPPED status:
> 
> > @umarcor:
> > Would it be acceptable to allow PASSED and/or SKIPPED to be passed to call_post_check instead?
> 
> At the moment, `call_post_check` is only executed if all the tests PASSED.
> 

I think post check should run if there are no FAILED tests. Thus it should run if there are SKIPPED tests as long as there are no FAILED tests. Currently there is no way to have SKIPPED tests without a FAILED test though.